src/backend/src/controllers/projects.controllers.ts

```diff
@@ -299,3 +299,72 @@ export const editProject = async (req: Request, res: Response) => {
   // return the updated work package
   return res.status(200).json(updatedProject);
 };
+
+export const setProjectTeam = async (req: Request, res: Response) => {
+  const { body } = req;
+
+  // check for valid WBS number
+  const parsedWbs: WbsNumber = validateWBS(req.params.wbsNum);
+
+  if (!isProject(parsedWbs)) {
+    return res.status(400).json({ message: `${req.params.wbsNum} is not a valid project WBS #!` });
+  }
+
+  const wbsEle = await prisma.wBS_Element.findUnique({
+    where: {
+      wbsNumber: {
+        carNumber: parsedWbs.carNumber,
+        projectNumber: parsedWbs.projectNumber,
+        workPackageNumber: parsedWbs.workPackageNumber
+      }
+    },
+    include: {
+      project: true
+    }
+  });
+
+  if (wbsEle === null) {
+    return res.status(404).json({ message: `project WBS ${req.params.wbsNum} not found!` });
+  }
+
+  // find the associated project
+  const project = await prisma.project.findFirst({
+    where: {
+      wbsElement: {
+        carNumber: parsedWbs.carNumber,
+        projectNumber: parsedWbs.projectNumber,
+        workPackageNumber: parsedWbs.workPackageNumber
+      }
+    }
+  });
+
+  if (!project) {
+    return res.status(404).json({ message: `no associated project for ${req.params.wbsNum}` });
+  }
+
+  // check for valid team
+  const team = await prisma.team.findUnique({ where: { teamId: body.teamId } });
+  if (!team) {
+    return res.status(404).json({ message: `team with id ${body.teamId} not found.` });
+  }
+
+  // check for user and user permission (admin, app admin, or leader of the team)
+  const user = await prisma.user.findUnique({ where: { userId: body.submitterId } });
```

Comment:
instead of having a submitterId in the body, we should use the getUser method from utils which checks for who is currently logged in which is more secure and uses that